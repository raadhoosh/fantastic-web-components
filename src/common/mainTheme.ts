import { Theme } from './styled';
import * as Color from 'color';
function darken(hex: string) {
    return Color(hex)
        .darken(0.5)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#0066b1',
        secondary_dark: darken('#0066b1'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#d4edda',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    bottom: {
        fontsize: '1rem',
        border: '5px',
    },
    darken: (_color?: string) => {
        return Color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return Color(hex).fade(a ? a : 0.5).string();
    },
};

export default mainTheme;
