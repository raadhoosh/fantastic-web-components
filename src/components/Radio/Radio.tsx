import styled from '../../common/styled';
import * as React from 'react';

interface IProps {
    children?: any;
    backgroundColor?: string;
    color?: string;
    margin?: string;
    small?: boolean;
    large?: boolean;
    noShadow?: boolean;
    theme?: any;
    id?: string;
    name?: string;
    label?: string;
    disabled?: boolean;
}

const EmRadio = styled('input')(
    (props: (IProps)) => {
        const css = {
            outline: 'none',
            ...[`;position: absolute;`],
            opacity: '0',
            cursor: 'pointer',
            height: '0',
            width: '0',
            ':checked ~ .checkmark': {
                backgroundColor: props.theme.color.primary,
                ':after': {
                    display: 'block'
                }
            }
        };
        return css;
    }
);
const EmLabel = styled('label')(
    (props: (IProps)) => {
        const css = {
            color: '#000',
            display: 'flex',
            alignItems: 'flex-start',
            cursor: 'pointer',
            paddingLeft: '35px',
            marginBottom: '12px',
            ...[`;position: relative;`],
            '.checkmark': {
                ...[`;position: absolute;`],
                top: '0',
                left: '0',
                height: '25px',
                width: '25px',
                borderRadius: '50%',
                backgroundColor: props.theme.color.light,
                '::after': {
                    content: '""',
                    ...[`;position: absolute;`],
                    display: 'none',
                    left: '9px',
                    top: '9px',
                    width: '8px',
                    height: '8px',
                    backgroundColor: 'white',
                    borderRadius: '50%'
                }
            }
        };
        return css;
    }
);
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const Radio = (props: IProps) => {

    return (
        <EmLabel theme={props.theme} >
            <EmRadio type="radio"
                theme={props.theme}
                disabled={props.disabled}
                name={props.name}
            />
            <span className="checkmark" />
            <span>
                {props.label}
            </span>
        </EmLabel>
    );
};

export default Radio;
