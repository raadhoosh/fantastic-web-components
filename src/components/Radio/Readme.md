Radio  example:

```js
<Radio name='example'  label='Simple example'/>
<br/>
<Radio name='example'  label='Simple example'/>
<br/>

<Radio  disabled label="Don't know (Disabled)"/>

```