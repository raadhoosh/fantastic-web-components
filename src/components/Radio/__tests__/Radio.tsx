import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import Radio  from '../Radio';
import mainTheme from '../../../common/mainTheme';

test('renders with correct styles', () => {
    const inp = renderer.create(<Radio  theme={mainTheme} disabled />).toJSON();
    expect(inp).toMatchSnapshot();
});
