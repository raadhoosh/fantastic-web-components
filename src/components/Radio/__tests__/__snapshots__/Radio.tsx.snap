// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<label
  className="css-189tkbf"
>
  <input
    className="css-wui0w3"
    disabled={true}
    name={undefined}
    type="radio"
  />
  <span
    className="checkmark"
  />
  <span />
</label>
`;
